modules/blockbuilder/blockbuilder.go

```diff
@@ -359,12 +361,22 @@ outer:
 				break outer
 			}
 
+			if maxBytesPerCycle > 0 && consumedBytes+recordSizeBytes >= maxBytesPerCycle {
```

Comment:
What do you think about moving this check after pushTraces? That means it is guaranteed to always allow at least 1 record.   I am thinking about the case where the max is set to a value less than the size of the next record, the check would prevent the BB from ever making progress.

Reply:
Makes sense to me :)